framework/include/relationshipmanagers/AugmentSparsityOnInterface.h

```diff
@@ -75,13 +75,6 @@ class AugmentSparsityOnInterface : public RelationshipManager
 protected:
   virtual void internalInitWithMesh(const MeshBase &) override;
 
-  /**
-   * The Mesh we're calculating on
-   */
-  const AutomaticMortarGeneration * _amg;
-
-  bool _has_attached_amg;
-
```

Comment:
how do this commit's changes address thread safety?

Reply:
We were reading from and writing to the `_amg` pointer and the `_has_attached_amg` boolean, which are shared pieces of data across threads, without any protection. So there would be data races for those. By making the `amg` pointer a stack variable, each thread now has it's own copy of the pointer and there are no races for reading/writing